### PR TITLE
Add Bean Validation and tests

### PR DESCRIPTION
**Pull request by keithdmoore**
This PR addresses the issue closes https://github.com/katharsis-project/katharsis-core/issues/20

### CODE REVIEW

File: spring-boot-simple-example/src/main/java/io/katharsis/example/springboot/simple/domain/mapper/ConstraintViolationExceptionMapper.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+			ErrorData error = ErrorData.builder()
+					.setStatus(String.valueOf(CONFLICT_409))
+					.setTitle(violation.getMessage())
+					.setSourcePointer(createSourcePointer(violation))
+					.build();
+
+			errors.add(error);
+		}
+
+		return ErrorResponse.builder().setStatus(CONFLICT_409).setErrorData(errors).build();
```
Comment (bahdvp) [inline spring-boot-simple-example/src/main/java/io/katharsis/example/springboot/simple/domain/mapper/ConstraintViolationExceptionMapper.java]: 400 makes more sense to me here (bad request over CONFLICT)

File: spring-boot-simple-example/src/main/java/io/katharsis/example/springboot/simple/domain/model/Task.java [hunk 2]
```diff
@@ -31,6 +34,9 @@
 
     @JsonProperty("my-name")
     private String name;
+    
+    @Size(max = 20, message="Description may not exceed {max} characters.")
```
Comment (bahdvp) [inline spring-boot-simple-example/src/main/java/io/katharsis/example/springboot/simple/domain/model/Task.java]: should be 200?

File: spring-boot-simple-example/src/test/java/io/katharsis/example/springboot/simple/SpringBootSimpleExampleApplicationTests.java [hunk 3]
```diff
@@ -1,18 +1,111 @@
 package io.katharsis.example.springboot.simple;
```
Comment (bahdvp) [inline spring-boot-simple-example/src/test/java/io/katharsis/example/springboot/simple/SpringBootSimpleExampleApplicationTests.java]: did you mean to commit this file?

**Comment by chb0github** [comment]
Definitely!

**Pull request merged**